pkg/services/pluginsintegration/clientmiddleware/logger_middleware.go

```diff
@@ -76,6 +76,16 @@ func (m *LoggerMiddleware) QueryData(ctx context.Context, req *backend.QueryData
 		return innerErr
 	})
 
+	if err != nil {
+		return resp, err
+	}
+
+	for refID, dr := range resp.Responses {
+		if dr.Error != nil {
+			m.logger.Error("Query response error", "refID", refID, "error", dr.Error)
```

Comment:
Should we perhaps somehow indicate this is coming from within query data response?

```suggestion
			m.logger.Error("Query data response with embedded error", "refID", refID, "error", dr.Error)
```

Reply:
I think `refID` log param is enough to indicate that the error is embedded in the response

Reply:
Cool. I'm considering including status and status source here as well. Going to try it out first

Reply:
I did change the message. WDYT?

Reply:
Looks good to me! :+1: